test_fms/random_numbers/test_random_numbers.F90

```diff
@@ -0,0 +1,279 @@
+program test_random_numbers
+
+use fms_mod, only: fms_init, fms_end
+use platform_mod, only: r4_kind, r8_kind
+use mpp_mod, only: mpp_error, fatal, stderr, mpp_npes, mpp_pe
+use fms_string_utils_mod, only: string
+use time_manager_mod, only: time_type, get_date, set_date, set_calendar_type, JULIAN
+use random_numbers_mod
+
+implicit none
+
+integer, parameter :: n_moments = 1000 !> Highest-order moment to test
+integer, parameter :: n0_1d = 2500 !> Initial length of 1D random sample vector
+integer, parameter :: n0_2d = 50 !> Initial dimensions of 2D random sample array
+
+integer, parameter :: seeds(0:*) = [0, -5, 3] !> Seed constants
+integer, parameter :: k = KIND_ !> Either r4_kind or r8_kind
+
+real(k), dimension(n_moments) :: moment_mu !> Expected moment values
+real(k), dimension(n_moments) :: moment_sigma !> Standard deviations of sample moments
+
+type(time_type) :: now !> Current date and time
+
+call fms_init
+
+if (mpp_npes() .ne. 4) then
+  call mpp_error(fatal, "The random_numbers unit test requires four PEs")
```

Comment:
Should it be FATAL or should it be a NOTE and then exit the program?

Reply:
What's the problem with using `FATAL`? I believe it should end up with four PEs any time it's run through the script, so this is really just a sanity check that shouldn't fail under normal circumstances.